lib/open_api_spex/test/test_assertions.ex

```diff
@@ -75,4 +116,77 @@ defmodule OpenApiSpex.TestAssertions do
   def assert_request_schema(value, schema_title, api_spec = %OpenApi{}) do
     assert_schema(value, schema_title, api_spec, :write)
   end
+
+  @doc """
+  Asserts that the response body conforms to the response schema for the operation with id `operation_id`.
+  """
+  @spec assert_operation_response(Plug.Conn.t(), String.t()) ::
+          no_return | Plug.Conn.t()
+
+  def assert_operation_response(conn, operation_id \\ nil)
+
+  # No need to check for a schema if the response is empty
+  def assert_operation_response(conn, _operation_id) when conn.status == 204, do: conn
+
+  def assert_operation_response(conn, operation_id) do
+    {spec, operation_lookup} = PutApiSpec.get_spec_and_operation_lookup(conn)
+
+    operation_id = operation_id || conn.private.open_api_spex.operation_id
+
+    case operation_lookup[operation_id] do
+      nil ->
+        flunk(
+          "Failed to resolve schema. Unable to find a response for operation_id: #{operation_id} for response status code: #{conn.status}"
+        )
+
+      operation ->
+        validate_operation_response(conn, operation, spec)
+    end
+
+    conn
+  end
+
+  if OpenApiSpex.OpenApi.json_encoder() do
+    @spec validate_operation_response(
+            Plug.Conn.t(),
+            Operation.t(),
+            OpenApi.t()
+          ) ::
+            term | no_return
+    defp validate_operation_response(conn, %Operation{operationId: operation_id} = operation, spec) do
+      content_type = Utils.content_type_from_header(conn)
+
+      resolved_schema =
+        get_in(operation, [
+          Access.key!(:responses),
+          Access.key!(conn.status),
+          Access.key!(:content),
+          content_type,
+          Access.key!(:schema)
+        ])
+
+      if resolved_schema == nil do
+        flunk(
+          "Failed to resolve schema! Unable to find a response for operation_id: #{operation_id} for response status code: #{conn.status} and content type #{content_type}"
+        )
+      end
+
+      body =
+        if String.match?(content_type, ~r/^application\/.*json.*$/) do
+          OpenApiSpex.OpenApi.json_encoder().decode!(conn.resp_body)
+        else
+          conn.resp_body
+        end
+
+      assert_raw_schema(
+        body,
+        resolved_schema,
+        spec
+      )
+    end
+  else
+    defp validate_operation_response(_conn, _operation, _spec) do
+      flunk("Unable to use assert_operation_response unless a json encoder is configured")
```

Comment:
This error message https://github.com/open-api-spex/open_api_spex/pull/545/files#diff-f30f8c2c6963f59d6e63acbef475794bdf7f98d078e384f2295c6c83a9fc9e81R44 suggests a solution so we should either mention here as well that they can add an encoder library needs to be added to the mix.